core/renderers/geras/info.js

```diff
@@ -113,8 +113,8 @@ Blockly.geras.RenderInfo.prototype.addInput_ = function(input, activeRow) {
         new Blockly.blockRendering.ExternalValueInput(this.constants_, input));
     activeRow.hasExternalInput = true;
   } else if (input.type == Blockly.DUMMY_INPUT) {
-    // Dummy inputs have no visual representation, but the information is still
-    // important.
+    activeRow.elements.push(
```

Comment:
High-level question: why add an element for the dummy input at all, given that it's zero width and not supposed to have padding around it, instead of just adding a .align property on the row and setting it based on the input (s) on that row?

Reply:
Although dummy inputs don't have a width, they do have a MIN_HEIGHT that should be taken into consideration (partially what this is tackling see https://github.com/google/blockly/issues/3341).
That being said, we could set a MIN_HEIGHT on the row if we see a dummy input.